core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -505,6 +510,23 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging {
     getAll.sorted.map{case (k, v) => k + "=" + v}.mkString("\n")
   }
 
+  def getDriverExtraClassPath() : Option[String] = {
+    // Concatenate common and driver classpath in that order
+    val commonExtraClassPath = getOption("spark.common.extraClassPath")
+    val driverExtraClassPath = getOption("spark.driver.extraClassPath")
+    Some((commonExtraClassPath ++ driverExtraClassPath).mkString(File.pathSeparator))
+      .filter(_.nonEmpty)
```

Comment:
To be pedantic, the filtering should be done earlier, otherwise:

    scala> Seq("", "").mkString(":")
    res0: String = :